### PR TITLE
Create user signup 13

### PR DESCRIPTION
**Pull request by colinksmith**
Branch to post user signup and prevent duplicate account names. Also set up Jest testing for this file.

### CODE REVIEW

File: src/dev-mongo.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+// This file runs a MongoDB server locally for development
+module.exports.run = async () => {
+  const dbPath = path.join(__dirname, '..', '.mongo');
+
+  if (!fs.existsSync(dbPath)) {
+    fs.mkdirSync(dbPath);
+  }
+
+  // start mongo
+  const mongod = await MongoMemoryServer.create({
```
Comment (GUIMess) [inline src/dev-mongo.js:16]: A very small nitpick and it might be this way for a reason. Maybe we change this to mongoD?

**Review by GUIMess — APPROVED** [review]
This is some sick ass code! I'm really impressed. Feel free to address my comment and merge!

File: src/dev-mongo.js (hunk 1 above)
Comment (db556b) [reply src/dev-mongo.js:16]: This should likely be mongoDB 

Wouldn't let that stop a merge though. File a bug on it

Comment (colinksmith) [reply src/dev-mongo.js:16]: Yeah I wasn't quite sure, several tutorials seems to have it as mongod so it seemed like some format I wasn't familiar with. I'd be okay renaming it to mongoServer as well, as that feels more descriptive.

**Pull request merged**